plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -388,6 +389,14 @@ fu_thunderbolt_device_write_firmware(FuDevice *device,
 		fu_progress_set_status(progress, FWUPD_STATUS_DEVICE_RESTART);
 	}
 
+	/* this is weird, but if we've just updated the firmware, we want to wait for the device
+	 * to "come back" after the controller is re-setup() and the 5 seconds has elapsed to
+	 * reprobe the retimer */
+	if (fu_device_has_internal_flag(device, FU_DEVICE_INTERNAL_FLAG_NO_AUTO_REMOVE)) {
+		g_warning("\n\nNO_AUTO_REMOVE, so waiting for replug\n\n");
```

Comment:
nit; but I don't like the wrapped double newline, it will already be red (from a warning).  Is that not enough?

Reply:
Ohh this was just for test -- sorry -- this shouldn't have made it into this PR :) 

This is what I sent to Intel:

_I spent a few hours thinking about what fwupd actually does when enumerating the retimer and then updating it.

startup:
 * probe host controller
 * setup host controller, reading the version
 * if hwid retimer-offline-mode then set device force-enumeration
 * if device has force-enumeration
  * set port offline and rescan
  * wait 5 seconds
  * set port online
  
after few seconds post setup:
 * a retimer appears
 * we get the retimer version
 * after a few more seconds the port gets onlined
 * the retimer disappears from udev
 * ...which fwupd deliberately ignores

on retimer firmware update:
 * we run composite_prepare which sets the parent port offline (and rescan)
 * we get the current NVM if allowed, and check it's compatible (this is what was 'decompressing' for 10 seconds... now disabled
 * we dd the data to the nvm
 * we authenticate the retimer
 * (what we added yesterday, but I'm sceptical it did anything) we wait for the device to re-appear
 * we run composite_cleanup which sets the parent port online
 * the retimer disappears
 * at this point we query the version of the retimer, which has the old firmware version
 
So, asking again -- what do I have to do exactly to make the retimer sysfs show the correct version? :)

I'm tempted to say we have to do something more in composite_cleanup -- i.e. would adding a rescan before setting the parent port online change the version in sysfs? If you can work out the minimal steps required in the console to make sysfs show the "correct, new" nvm_version that would be great. It might be that the sysfs attribute for the nvm_version is updated after we do the authenticate (without a rescan) and if so the fix is even simpler._